mesh_database_client/database.py

```diff
@@ -0,0 +1,148 @@
+from dotenv import load_dotenv
+from numpy import sqrt
+
+load_dotenv()
+
+class DatabaseClient:
+
+    def __init__(self, spreadsheet_id = None):
+
+        if spreadsheet_id is None:
+            self.spreadsheet_id = os.environ.get("SPREADSHEET_ID")
+        else:
+            self.spreadsheet_id = spreadsheet_id
+
+        SCOPES = ['https://www.googleapis.com/auth/spreadsheets.readonly']
+        # created in Google Cloud admin
+        SERVICE_ACCOUNT_FILE = 'credentials.json'
+
+        signup_sheet_range = 'Form Responses 1!A:AO'
+        links_sheet_range = 'Links!A:I'
+
+        CREDS = service_account.Credentials.from_service_account_file(
+        SERVICE_ACCOUNT_FILE, scopes=SCOPES)
+
+        self.service = build('sheets', 'v4', credentials=CREDS)
+
+        # Call the Sheets API
+        self.sheet = self.service.spreadsheets()
+
+        # load all data into dataframe
+        self.signup_df = self.get_range_as_df(signup_sheet_range)
+        self.links_df = self.get_range_as_df(links_sheet_range)
+
+        self.process_signup_df()
+        self.process_links_df()
+
+        # drop last signup_df column which does not contain data
+        self.signup_df.drop(self.signup_df.tail(1).index,inplace=True)
+
+        self.gmaps = googlemaps.Client(key=os.environ.get("MAPS_API"))
+
+    def get_range_as_df(self, range):
+        result = self.sheet.values().get(spreadsheetId=self.spreadsheet_id,range=range).execute()
+        values = result.get('values', [])
+        
+        df = pd.DataFrame(values[1:], columns = values[0])
+        return df
+
+    def process_signup_df(self):
+        df = self.signup_df
+ 
+        # force columns to be specific type
+        df['NN'] = (pd.to_numeric(df['NN'], errors="coerce")
+                         .fillna(0)
+                         .astype(int))
+
+        df['Latitude'] = pd.to_numeric(df['Latitude'], errors='coerce')
+        df['Longitude'] = pd.to_numeric(df['Longitude'], errors='coerce')
+         
+        self.signup_df = df
+
+
+    def process_links_df(self):
+        df = self.links_df
+
+        # force columns to be specific type
+        df = df.astype({'to':'float'})
+        df = df.astype({'from':'float'})
+        df = df.astype({'to':'int'})
+        df = df.astype({'from':'int'})
+        
+        self.links_df = df
+
+    def name_to_nn(self, name):
+        signup_df = self.signup_df
+        name_df = signup_df[signup_df['Name'].str.contains(name, case=False)]
+        if not name_df.empty:
+            entry = name_df.iloc[0]
+            return entry['NN']
+        else:
+            return None
+
+    def email_to_nn(self, email):
+        # TODO multiple emails recency
+        signup_df = self.signup_df
+        email_df = signup_df[signup_df['Email'].str.contains(email, case=False)]
+        if not email_df.empty:
+            entry = email_df.iloc[0]
+            return entry['NN']
+        else:
+            return None
+
+    def deg_to_feet(deg):
```

Comment:
I think you want to add ` @staticmethod` here. Currently `deg` will be set to a class reference (like `self`).

Reply:
Or remove it. Seems unused 😉

Reply:
Yep, unused!